From e445814d11b8c7520a15035387b511ff2da0c0a2 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Thu, 28 Nov 2024 10:16:25 +0100
Subject: [PATCH] media: stm32: dcmipp: set V4L2_SUBDEV_FL_STREAMS for
 dcmipp-input

dcmipp-input must use the stream apis in order to be able to properly
perform the selection of input data and perform the routing into pipes.

Change-Id: I87874b840b56957328e55f186c6ee8ce0db72893
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/499331
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Alain VOLMAT <alain.volmat@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-common.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-common.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-common.c
index bf6203c9116e..bdaceae8f990 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-common.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-common.c
@@ -72,6 +72,9 @@ int dcmipp_ent_sd_register(struct dcmipp_ent_device *ved,
 	sd->flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
 	if (sd->ctrl_handler)
 		sd->flags |= V4L2_SUBDEV_FL_HAS_EVENTS;
+	/* dcmipp_input deals with streams to perform proper routing */
+	if (!strcmp(name, "dcmipp_input"))
+		sd->flags |= V4L2_SUBDEV_FL_STREAMS;
 
 	/* Initialize the media entity */
 	ret = media_entity_pads_init(&sd->entity, num_pads, ved->pads);
-- 
2.53.0

